app/routes/task.py

```diff
@@ -0,0 +1,125 @@
+from app import db
+from app.models.task import Task
+from flask import Blueprint, jsonify, request, make_response, abort
+from datetime import datetime as dt
+import requests
+import os
+
+bp = Blueprint("task_bp", __name__, url_prefix="/tasks")
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"details":f"{cls.__name__} {model_id} invalid"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"details":f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model
+
+@bp.route("", methods=["POST"])
+def create_task():
+    try:
+        request_body = request.get_json()
+        new_task = Task.from_dict(request_body)
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        task_dict = new_task.to_dict()
+
+        return make_response(jsonify({
+            "task": task_dict}), 201)
+            
+    except:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+@bp.route("", methods=["GET"])
+def read_all_tasks():
+    sort_query = request.args.get("sort")
+    task_query = Task.query
+    if sort_query == "asc":
+        task_query = Task.query.order_by(Task.title.asc())
+    if sort_query == "desc":
+        task_query = Task.query.order_by(Task.title.desc())
+    tasks = task_query.all()
+
+    tasks_response = [task.to_dict() for task in tasks]
+    
+    return jsonify(tasks_response), 200
+
+@bp.route("/<task_id>", methods=["GET"])
```

Comment:
In the `goal` routes, you left off the leading `/`, so for consistency, I'd recommend following the same pattern (or add the leading slashes where applicaple in the goal routes.